torchtune/modules/transformer.py

```diff
@@ -145,20 +290,32 @@ class TransformerDecoder(nn.Module):
     def __init__(
         self,
         tok_embeddings: nn.Embedding,
-        layer: TransformerDecoderLayer,
-        num_layers: int,
+        layers: Union[nn.Module, List[nn.Module]],
+        num_layers: Optional[int],
         max_seq_len: int,
         num_heads: int,
         head_dim: int,
         norm: nn.Module,
         output: nn.Linear,
+        output_hidden_states: Optional[List[int]] = None,
```

Comment:
tiny nit, why not?
```suggestion
        output_hidden_states: Optional[List[int]] = [],
```

Reply:
You don't want to use mutable types as a default value in Python. Otherwise, all instances of TransformerDecoder that use the default value will actually share the same list object. Since we don't modify the list here in the class, it's not actually a problem, but it's a risky pattern nonetheless that is best avoided.